### PR TITLE
Mashraf/audio streamer

### PR DESCRIPTION
**Pull request by 22mahmoud**

### CODE REVIEW

File: src/helpers/AudioStreamer.ts [hunk 1]
```diff
@@ -1,141 +1,84 @@
...
+    // tslint:disable-next-line no-shadowed-variable
+    const AudioContext: AudioContext =
+      window.AudioContext || window.webkitAudioContext;
+    this.context = new AudioContext();
+    this.source = this.context.createMediaStreamSource(stream);
     /*
-     * @param {function} onData Callback to run on data each time it's received
-     * @param {function} onError Callback to run on an error if one is emitted.
+     * TODO: don't use createScriptProcessor becuse of it is deprecated ⚠️
+     *	We Should use audioworklet insted
```
Comment (piraka9011) [inline src/helpers/AudioStreamer.ts]: `AudioWorklet` is apparently experimental in Firefox, is this mostly for compatibility with chrome?
https://developer.mozilla.org/en-US/docs/Web/API/AudioWorklet

File: src/helpers/AudioStreamer.ts [hunk 2]
```diff
@@ -1,141 +1,84 @@
...
-    }
-    if (context) {
-      context.close().then(() => {
-        input = null;
-        processor = null;
-        context = null;
-        AudioContext = null;
-      });
+    } catch (error) {
+      console.log('ERROR', error.toString());
```
Comment (piraka9011) [inline src/helpers/AudioStreamer.ts]: `console.error()` 🤷‍♂ ?

Comment (22mahmoud) [reply src/helpers/AudioStreamer.ts]: Ahh  I forgot to remove it  :smile:

File: config/values.js [hunk 3]
```diff
@@ -245,7 +245,7 @@ const values = {
       // specify the inclusion of additional modules below.
       devVendorDLL: {
         // Enabled?
-        enabled: true,
+        enabled: false,
```
Comment (Ananasship) [inline config/values.js]: Does webpack throw an error when it's turned to `true`?

File: src/helpers/AudioStreamer.ts [hunk 4]
```diff
@@ -1,141 +1,84 @@
...
-  initRecording(onData, onError) {
+
+  start() {
+    navigator.mediaDevices
+      .getUserMedia({ audio: true })
+      .then(this.handleStream)
+      .catch(this.errorHandler);
+  }
+
+  handleStream(stream: MediaStream) {
```
Comment (Ananasship) [inline src/helpers/AudioStreamer.ts]: Can we add a small comment here explaining what this function does

File: src/pages/Recognition/index.tsx [hunk 5]
```diff
@@ -55,100 +55,117 @@ class Recognition extends React.Component<IProps, IState> {
     showErrorMessage: false,
     errorMessage: '',
   };
-  handleRecordingButton = () => {
+
+  handleRecordingButton = async () => {
```
Comment (Ananasship) [inline src/pages/Recognition/index.tsx]: nice

File: config/values.js (hunk 3 above)
Comment (piraka9011) [reply config/values.js]: For me it still does :/

File: src/helpers/AudioStreamer.ts (hunk 4 above)
Comment (22mahmoud) [reply src/helpers/AudioStreamer.ts]: done

**Pull request merged**